src/components/settings/SettingsAccessTab/SettingsAccessGenerateLinkButton.jsx

```diff
@@ -57,10 +57,11 @@ const SettingsAccessGenerateLinkButton = ({
 
 SettingsAccessGenerateLinkButton.defaultProps = {
   disabled: false,
+  formattedLinkExpirationDate: null,
 };
 SettingsAccessGenerateLinkButton.propTypes = {
   enterpriseUUID: PropTypes.string.isRequired,
-  formattedLinkExpirationDate: PropTypes.string.isRequired,
+  formattedLinkExpirationDate: PropTypes.string,
```

Comment:
[curious] Should this be allowed to be `null` if `expiration_date` is a required field on the `EnterpriseCustomerInviteKey` model ([source](https://github.com/openedx/edx-enterprise/blob/master/enterprise/models.py#L3323-L3329))?

Reply:
My original thought was the disabled should always = true if formattedLinkExpirationDate is null. But I think we can enforce it in this component so that the button is disabled if formattedLinkExpirationDate is null.

Reply:
Got it, thanks for the clarification. Makes sense that we'll disable the button when formattedLinkExpirationDate is null. 

Currently, though, this component is largely controlled in that it's solely up to its parent to control its disabled behavior and would probably prefer not to mix controlled with uncontrolled component usage here. That is, still rely solely on the parent to control whether this button is disabled or not (based on a null `formattedLinkExpirationDate`). Just worried that we're not mixing the `disabled` behavior to not always be based on what's passed into the `disabled` prop itself from the parent component.

Could we instead make the `formattedLinkExpirationDate` prop conditionally required if `disabled` is false?

See [this example](https://github.com/openedx/paragon/blob/a95624d0721dcfddb7a201543fb23cf44f07c931/src/utils/propTypes/utils.js#L51) of a custom prop type util function to do this type of conditional prop checking from Paragon and [these docs](https://reactjs.org/docs/typechecking-with-proptypes.html) for more details on creating custom validators.

Reply:
That is a good point, let me look into this.